docs/DeveloperGuide.md

```diff
@@ -1,29 +1,115 @@
+
+#### Delete Command
+
+Delete command is used to delete an item from the packing list.
+
+Mechanism: ```DeleteCommand.execute()``` calls the ```PackingList.deleteItem()``` method from the ```PackingList``` class which executes the ```ArrayList.remove()``` method to remove the item from the ```PackingList``` ArrayList.
+
+
+#### Help Command
+Help command is used to exit the BagPacker application.
+
+Execute: ```HelpCommand.execute()``` prints the following help message.
+
+```
+All Commands:
+1. add : Adds a quantity of items to the packing list.
+	Example: add 3 toothbrushes
+2. delete : Deletes an item from the packing list.
+	Example: delete 1
+3. list : List all items in packing list.
+	Example: list
+4. pack : Marks an item as packed in the packing list.
+	Example: pack 2 of 3
+	Meaning: packs 2 quantities of the third item in the packing list
+5. unpack : Marks an item as unpacked in the packing list.
+	Example: unpack 1 of 2
+	Meaning: unpacks 1 quantity of the second item in the packing list
+6. bye : Stops the BagPacker Application
+	Example: bye
+____________________________________________________________
+
+```
+
+#### Bye Command
+```ByeCommand``` is used to exit the BagPacker application.
+
+Mechanism: ```ByeCommand.execute()``` updates the static boolean ```isBagPackerRunning``` to be false. 
+The ```runBagPacker()``` method will continually parse and execute relevant commands (refer to Command Mechanisms in DG) until
+```isBagPackerRunning == false``` which occurs upon the execution of the ```byeCommand```.
+
+#### DeleteList Command
+```DeleteListCommand``` is used to delete a whole packing list in the BagPacker application.
+
+Mechanism: ```DeleteListCommand.execute()``` reassigns the existing ```packingList``` to a new empty ArrayList of Items, thus deleting the ```packingList```.
+
 
 
 ## Product scope
+
 ### Target user profile
 
-{Describe the target user profile}
+BagPacker is for NUS students, in particular, exchange students who travel a lot and want a simple CLI to keep track of their packing.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+BagPacker aims to help busy students simplify their packing process by allowing easy adding of items to pack and record of the items they have already packed so that they can be organised and aboard their travels with ease.
+
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...                                  | So that I can ...                                                            |
```

Comment:
Good job with specifying the user stories by version! Gives a good understanding of the actual program and the progress made